### PR TITLE
Sync FreeBSD changes

### PR DESCRIPTION
**Pull request by jlduran**

### CODE REVIEW

**Comment by jlduran** [comment]
If this pull request is accepted, a drop-in replacement of dtc can be applied to the FreeBSD sources.
If it is not too much to ask, I would like it to happen before 14.0-RELEASE. (cc/ @emaste)

**Review by emaste — APPROVED** [review]

File: input_buffer.cc [hunk 1]
```diff
@@ -39,11 +39,11 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <limits>
 #include <functional>
 #ifndef NDEBUG
 #include <iostream>
 #endif
+#include <limits>
```
Comment (emaste) [inline input_buffer.cc:46]: commit message doesn't seem to match?

Comment (jlduran) [reply input_buffer.cc:46]: Yes and no, FreeBSD added the change in https://github.com/freebsd/freebsd-src/commit/1c44d2bf86834a7214e0cc7bde7c956b5b02fd21.
It is only partially applied here, and I left the commit messages verbatim.

Comment (jlduran) [reply input_buffer.cc:46]: Let me know if I should add an `Obtained from:` entry in each commit message.

**Review by davidchisnall — APPROVED** [review]

**Pull request merged**